STM32F405RGTx/Core/Src/rtos_threads/measurements_thread.cpp

```diff
@@ -39,14 +45,51 @@ void MeasurementsThread::processData() {
         uint32_t val = (sum/ADC_DECIMATION_COEFF);
 
         switch (i) {
-            // TODO need to figure out which variables need to be loaded for each 'i'
-            // global_mc_data.pVa = ...;
-            // global_mc_data.pVb = ...;
-            // global_mc_data.pVc = ...;
+
+            // read and convert phase voltages and dc voltage
             case 0:
-                printf("Val = %lx\r\n", val);
+               g_mc_data.pVa = VOLTAGE_DIVIDER_CONVERSION( ADC_TO_VOLTAGE( val ) );
+            case 1:
+               g_mc_data.pVb = VOLTAGE_DIVIDER_CONVERSION( ADC_TO_VOLTAGE( val ) );
+            case 2:
+               g_mc_data.pVc = VOLTAGE_DIVIDER_CONVERSION( ADC_TO_VOLTAGE( val ) );
+            case 3:
+               g_mc_data.dc_voltage = VOLTAGE_DIVIDER_CONVERSION( ADC_TO_VOLTAGE( val ) );
+
+            // read and convert phase currents
+            case 4:
+               g_mc_data.pIa = VOLTAGE_TO_CURRENT( val );
+            case 5:
+               g_mc_data.pIb = VOLTAGE_TO_CURRENT( val );
+            case 6:
+               g_mc_data.pIc = VOLTAGE_TO_CURRENT( val );
+
+            // read and convert adc temperatures
+            // NOTE: the ordering might change depending physical arrangement later on
+            case 7:
+            // Note: dc_cap_temp dne for Powerboard rev 2 but will for the next rev
+               val = ADC_TO_VOLTAGE(val);
+               val = VOLTAGE_TO_RESISTANCE(val);
```

Comment:
Again, these conversions should be done by the LUT.

If you think about what's happening in this code, `ADC_TO_VOLTAGE` will return a float, since its a voltage (although you are implicitly truncating the decimals since `val` is declared as an integer). Then you are trying to index an array with a float. Does this make sense?

The conversions from ADC to voltage, and voltage to resistance, then finally resistance to temperature ALL needs to be done in the lookup table, if we're going to be using a lookup table for this (which we should be imo).